server.js

```diff
@@ -1,44 +1,80 @@
-let express = require('express')
-data = require("./data/weather.json")
+const express = require('express');
+const cors = require("cors");
+const app = express();
+const axios = require("axios");
+app.use(cors());
+require("dotenv").config();
+app.use(express.json());
+
+const port = 3300
 
-let app = express()
-let port = 3300
+data = require("./data/weather.json")
 
-class Weather {
-    constructor(weatherObj) {
-        this.cityName = weatherObj.cityName;
+class Forecast {
+    constructor(valid_date, description) {
+        this.date = valid_date;
+        this.description = description;
     }
 }
 
 app.get('/', (request, response) => {
-  response.send('Welcome To The Weather!')
+  response.send('City Explorer Weather!')
 })
 
-app.get('/weather', (request, response) => {
-    let weatherState = request. query.weatherState;
-    let infoOfState = data
-    let cityName = request.query.cityName;
-    
-    let weather = data.find((w) => w.cityName === cityName)
-    if(weather) {
-        response.status(200).send(new Weather(weather));
-    } else {
-        response.status(404).send(
-            {
-                error: "City not found"
-            }
-        );
+app.get('/weather', async (request, response) => {
+    const cityLat = req.query.lat;
+    const cityLon = req.query.lon;
+    const searchQuery = req.query.searchQuery;
```

Comment:
When you declared your route you used request so those reqs need to match to work!